src/imitation/algorithms/bc.py

```diff
@@ -142,26 +142,6 @@ def __call__(
         )
 
 
-@dataclasses.dataclass(frozen=True)
```

Comment:
Why do we need to remove this? It seems largely unrelated to the PR. I don't have strong preferences on whether to keep it or not in general -- for reference https://github.com/HumanCompatibleAI/imitation/pull/422 is where we introduced it and tagging @ernestum who may have ideas on whether we should keep this and, if so, how to refactor it.

Reply:
It's possible to refactor, but I think the main reason I inlined the logic is to keep the `optim.step()` line close to the one for the incomplete batch, so you can see them right next to each other.